libs/wingc/src/type_check/jsii_importer.rs

```diff
@@ -148,107 +159,160 @@ impl<'a> JsiiImporter<'a> {
 		}
 	}
 
-	fn lookup_or_create_type(&mut self, type_fqn: &String) -> TypeRef {
-		let type_name = &self.strip_assembly_from_fqn(type_fqn);
-		// Check if this type is already define in this module's namespace
-		if let Ok(t) = self.env.lookup_nested_str(type_name, true, None) {
-			return t.as_type().expect(&format!("Expected {} to be a type", type_name));
+	fn lookup_or_create_type(&mut self, type_fqn: &FQN) -> TypeRef {
+		// Check if this type is already imported
+		if let Ok(t) = self
+			.wing_types
+			.libraries
+			.lookup_nested_str(type_fqn.as_str(), true, None)
+		{
+			return t.as_type().expect(&format!("Expected {} to be a type", type_fqn));
 		}
 		// Define new type and return it
 		self.import_type(type_fqn);
 		self
-			.env
-			.lookup_nested_str(type_name, true, None)
-			.expect(&format!("Expected {} to be defined", type_name))
+			.wing_types
+			.libraries
+			.lookup_nested_str(type_fqn.as_str(), true, None)
+			.expect(&format!("Expected {} to be defined", type_fqn))
 			.as_type()
 			.unwrap()
 	}
 
-	fn fqn_to_type_name(&self, fqn: &str) -> String {
-		let parts = fqn.split('.').collect::<Vec<&str>>();
-		let assembly_name = parts[0];
-		let namespace_name = parts[1];
-		fqn
-			.strip_prefix(format!("{}.{}.", assembly_name, namespace_name).as_str())
-			.unwrap()
-			.to_string()
-	}
+	fn import_type(&mut self, type_fqn: &FQN) {
+		// Hack: if the class name is a construct base then we treat this class as a resource and don't need to define it
+		if type_fqn.is_construct_base() {
+			return;
+		}
 
-	fn strip_assembly_from_fqn(&self, fqn: &str) -> String {
-		let parts = fqn.split('.').collect::<Vec<&str>>();
-		let assembly_name = parts[0];
-		fqn
-			.strip_prefix(format!("{}.", assembly_name).as_str())
-			.unwrap()
-			.to_string()
-	}
+		self.setup_namespaces_for(&type_fqn);
 
-	fn import_type(&mut self, type_fqn: &str) {
-		// Hack: if the class name is a construct base then we treat this class as a resource and don't need to define it
-		if Self::is_fqn_construct_base(type_fqn) {
+		// Check if this is a JSII interface and import it if it is
+		let jsii_interface = self.jsii_types.find_interface(type_fqn.as_str());
```

Comment:
Shouldn't `find_interface()` and `find_class()` take an `FQN`?

```suggestion
		let jsii_interface = self.jsii_types.find_interface(type_fqn);
```

Reply:
These are methods from `wingii` -- at some point it probably makes sense to use this `FQN` struct there and just import / use that type in `wingc`, but I decided against doing that work it in this PR because that library leverages code generation so it might be nontrivial.